contracts/staking.sol

```diff
@@ -116,4 +136,27 @@ contract Staking {
         _addressToValidatorIndex[staker] = 0;
         _validators.pop();
     }
+
+    function _updateMinimumStakedAmount() private {
+        if (_validators.length == 0) {
+            return;
+        }
+
+        uint256 min = _addressToStakedAmount[_validators[0]];
+        _lowestValidator = _validators[0];
+
+        for (uint32 i = 1; i < _validators.length; i++) {
+            if (_addressToStakedAmount[_validators[i]] < min) {
+                min = _addressToStakedAmount[_validators[i]];
+                _lowestValidator = _validators[i];
+            }
+        }
+        _minimumStakedAmountByValidator = min;
+    }
```

Comment:
Scanning all validators every time an account stake or unstake looks not efficient in terms of gas usage.
There are 2 approaches, I think
(1) Contract has MinimumStakeAmountByValidator and its address already. I think you can just compare with this.
(2) You can use some data structure like heap to reduce computation
cc: @zivkovicmilos